update developer docs for 1proposal flow

LaraAS opened the pull request:
Update the developer documentation to the new 1proposal flow. This PR should only be merged if and when the NNS decides to enable the new 1proposal flow.

LaraAS commented on docs/developer-docs/integrations/sns/tokenomics/sns-checklist.md:
```diff
@@ -139,7 +122,7 @@ It’s strongly recommended that developers have a discussion in the forum with
 It is recommended to share:
 
-- [x] The init file and swap file (example: [OpenChat](https://github.com/open-ic/open-chat/tree/master/sns/config)). <!-- NEW: init file-->
+- [x] The init file and swap file (example: [OpenChat](https://github.com/open-ic/open-chat/tree/master/sns/config)). <!-- NEW: TODO init file-->
```
@DanielThurau do we happen to have a new example file here? If not, I can just remove this line..

LaraAS replied:
Also, @DanielThurau, we don't have a swap file, right? I would remove this to just say "the init file" (used to say "the init file and swap file"). please let me know if this is wrong

DanielThurau replied:
Confirmed we do not have a swap file and you should delete it. I am currently working on a template that we can include in the docs of the new configuration file. I will have that ready for you in the next couple days.

DanielThurau replied:
Here is the link https://github.com/dfinity/ic/blob/master/rs/sns/cli/sns_init_template.yaml

LaraAS replied:
Linked to it, thanks!

LaraAS commented on docs/developer-docs/integrations/sns/tokenomics/preparation.md:
```diff
@@ -8,44 +8,30 @@ for examples using the
+All SNS parameters are set in a **.yaml** file that can then be passed as an argument to the NNS proposal that triggers the SNS launch, both for testing and in production.
+There are some parameters that have to be set and others that are set to a default value but that can also be changed. To make sure that all parameters are set to valid values, that are also consistent with each other, there is a tool to validate a given input file.
 
 **To create the SNS parameter yaml file and validate it, follow the steps
```
@LaraAS @DanielThurau could you confirm if this is still the case please?

DanielThurau replied:
This is where I recommend we link the template file that I create from this ticket https://dfinity.atlassian.net/browse/NNS1-2385

DanielThurau replied:
We do not have a generate the config file via that readme yet.

domwoe commented on docs/developer-docs/integrations/sns/tokenomics/sns-checklist.md:
```diff
@@ -13,11 +13,12 @@ launch – that is a decision by the NNS DAO. Ultimately, following this guide a
 :::
 
-<!-- NEW: [INFO BOX STYLE] 
+::: info
```
Remove whitespace (:::info)

LaraAS replied:
Good catch, thanks!

LaraAS replied:
Thanks. Changed in next commit

domwoe commented on docs/developer-docs/integrations/sns/tokenomics/sns-checklist.md:
```diff
@@ -57,46 +58,28 @@ It may not be possible to fully decentralize the dapp at the time of the SNS lau
+ As a part of the SNS launch process, an NNS proposal must be created to ask the NNS to create an SNS and start a decentralization swap. You can find the details about how to create such a proposal [here](../launching/launch-summary-1proposal.md).
 
-<!-- NEW: REMOVE-->
+Include relevant information like tokenomics (token distribution, governance, decentralization swap), details about the dapp (link to the open sourced code), whitepaper and anything else relevant for the community to make a decisions if an SNS should be launched for the dapp. 
```
I suggest to also explicitly mention the link to the deliberation forum post.

LaraAS replied:
This actually only comes later (section 3.2.) I can add a forward link, but in the first few sections it is not at all discussed where all of this information is shared. 
Maybe we should consider changing that and do this in a separate MR?

LaraAS replied:
I added a forward link to the forum and mentioned the forum discussion. 
I propose that we make further improvements in separate PRs.

domwoe commented on docs/developer-docs/integrations/sns/tokenomics/sns-checklist.md:
```diff
@@ -57,46 +58,28 @@ It may not be possible to fully decentralize the dapp at the time of the SNS lau
-A 1-proposal process is planned to be rolled out in August 2023. After this, the SNS launch will only require one NNS proposal.
-:::
-<!-- NEW: REMOVE-->
+### 1.5.1. More information
```
Since we now have only one subsection, it makes sense to remove subsection headings completely.

LaraAS replied:
done

domwoe commented on docs/developer-docs/integrations/sns/tokenomics/sns-checklist.md:
```diff
@@ -139,7 +122,7 @@ It’s strongly recommended that developers have a discussion in the forum with
 
 It is recommended to share:
 
-- [x] The init file and swap file (example: [OpenChat](https://github.com/open-ic/open-chat/tree/master/sns/config)). <!-- NEW: init file-->
```
What actually is the swap file?

LaraAS replied:
I don't know :D 
I assume this meant to say the information that you put into the openSwap proposal, but I am not aware of any file

LaraAS replied:
will clarify and remove this part

DanielThurau commented on docs/developer-docs/integrations/sns/launching/index.md:
```diff
@@ -1,18 +1,13 @@
-Then, it explains the practical [commands & actions to go through SNS launch](./launch-steps.md), in terms of concrete tools and commands that are needed to take a project through all of these stages and launch an SNS for it.
-
-<!-- NEW: 
 It starts with a detailed description of all Stages of an SNS launch.
```
```suggestion
It starts with a detailed description of all stages of an SNS launch.
```
?

LaraAS replied:
yes. applying.

DanielThurau commented on docs/developer-docs/integrations/sns/launching/index.md:
```diff
@@ -1,18 +1,13 @@
 Then, it explains the practical commands & actions to go through SNS launch, in terms of concrete tools and commands that are needed to take a project through all of these stages and launch an SNS for it.
 
-On August 14th, 2023, the NNS has decided to release a new version of the SNS with a new, simplified launch process. In particular, the new launch process only requires one single NNS proposal that will, if adopted, trigger the creation of the SNS canisters as well as starting and finishing the decentralization swap. Since the new flow is considerably simpler both for developers who want to hand  ...
+In August 2023, the NNS has decided to release a new version of the SNS with a new, simplified launch process. In particular, the new launch process only requires one single NNS proposal that will, if adopted, trigger the creation of the SNS canisters as well as starting and finishing the decentralization swap. Since the new flow is considerably simpler both for developers who want to hand over t ...
```
```suggestion
In August 2023, the NNS voted to release a new version of the SNS with a new, simplified launch process. In particular, the new launch process only requires one single NNS proposal that will, if adopted, trigger the creation of the SNS canisters as well as starting and finishing the decentralization swap. Since the new flow is considerably simpler both for developers who want to hand over their dapp to the IC and for voters/users, it is recommended to use the new flow.
```

DanielThurau commented on docs/developer-docs/integrations/sns/launching/index.md:
```diff
@@ -1,18 +1,13 @@
-The pages [Stages of an SNS launch](./launch-summary.md) and [Commands & actions to go through SNS launch](./launch-steps.md) define the old, legacy flow that might be deprecated going forward.
--->
\ No newline at end of file
+Therefore, the developer documentation currently includes pages to describe both the legacy and the new flow.
+The pages [Stages of an SNS launch](./launch-summary-1proposal.md) and [Commands & actions to go through SNS launch](./launch-steps-1proposal.md) define the new, recommended 1-proposal launch flow and how to go through it.
+The pages [Stages of an SNS launch](./launch-summary.md) and [Commands & actions to go through SNS launch](./launch-steps.md) define the old, legacy flow that might be deprecated going forward.
```
```suggestion
Therefore, the developer documentation currently includes pages to describe both the legacy and the new flow.
The pages [Stages of an SNS launch](./launch-summary-1proposal.md) and [Commands & actions to go through SNS launch](./launch-steps-1proposal.md) define the new, recommended 1-proposal launch flow and how to go through it.
The pages [Stages of an SNS launch - legacy flow](./launch-summary.md) and [Commands & actions to go through SNS launch - legacy flow](./launch-steps.md) define the old, legacy flow that might be deprecated going forward.
```

LaraAS replied:
good idea. will do

DanielThurau commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -30,106 +30,95 @@ These scripts have been most tested with a stack that looks like this:
         <td>0</td>
         <td>Developers deploy a dapp to the Internet Computer</td>
-        <td><code>./deploy_test_canister.sh</code></td>
+        <td><code> TODO ./deploy_test_canister.sh</code></td>
```
This is still correct

DanielThurau commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -30,106 +30,95 @@ These scripts have been most tested with a stack that looks like this:
         <td>1</td>
         <td>Dapp developers choose the initial parameters of the SNS for a dapp</td>
-        <td rowspan="4"><code>./deploy_sns.sh</code></td>
+        <td rowspan="2"><code>TODO ./deploy_sns.sh</code></td>
```
```suggestion
        <td rowspan="2"><code>./propose_sns.sh</code></td>
```

DanielThurau commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -30,106 +30,95 @@ These scripts have been most tested with a stack that looks like this:
     <tr>
         <td>1</td>
         <td>Dapp developers choose the initial parameters of the SNS for a dapp</td>
-        <td rowspan="4"><code>./deploy_sns.sh</code></td>
+        <td rowspan="2"><code>TODO ./deploy_sns.sh</code></td>
     </tr>
     <tr>
         <td>2</td>
-        <td>Dapp developers submit NNS proposal so they can deploy to the SNS subnet</td>
+        <td>Submit NNS proposal to create SNS</td>
     </tr>
     <tr>
         <td>3</td>
-        <td>Proposal #1 (of 3) is passed or rejected</td>
+        <td>Dapp developers add NNS root as co-controller of dapp</td>
+        <td><code>TODO</code></td>
+        <td><code></code></td>
     </tr>
```
Stages 3 needs to be switched with stage 1 and 2. The nns root has to be in control of the dapps before submitting the proposal

LaraAS replied:
Out of curiosity: is this a technical requirement or just a recommendation?
In practice, I think I would recommend to switch 2 and 3 (the choice of the init file should be made and everything should be ready by the time you hand over the SNS to NNS root). WDYT? I'll go ahead and make a proposal in line with this.

DanielThurau replied:
SGTM.

DanielThurau commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -30,106 +30,95 @@ These scripts have been most tested with a stack that looks like this:
         <td>3</td>
-        <td>Proposal #1 (of 3) is passed or rejected</td>
+        <td>Dapp developers add NNS root as co-controller of dapp</td>
+        <td><code>TODO</code></td>
```
```suggestion
        <td><code>./let_nns_control_dapp.sh</code></td>
```

DanielThurau commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -30,106 +30,95 @@ These scripts have been most tested with a stack that looks like this:
         <td>4</td>
-        <td>Dapp developers trigger the SNS canisters to be created on SNS subnet</td>
+        <td>The NNS proposal is decided</td>
+        <td rowspan="7"><code>TODO ./deploy_sns.sh</code></td>
```
The way I would represent this is with stages 4-8 should have no script and it should be "automated". There should be a step in between 8 and 9, which is `participate in the swap` and the code link is `./participate_in_sns_swap.sh`. then the current step 9 should also have no code link and be "automated"

LaraAS replied:
I now added a step as you suggested but I did not add a number.
This way, the layout here would match what you said but still the "Stages" in the Launch consdier "what devs do / what the system does" and also it is a bit less error prone to not change numbering now.
let me know if you disagree with that.

DanielThurau replied:
🙏

DanielThurau commented on docs/developer-docs/integrations/sns/tokenomics/preparation.md:
```diff
@@ -8,44 +8,30 @@ for examples using the
-To make sure that all parameters are set to valid values, that are also consistent
-with each other, there is a tool to validate this input file. 
+All SNS parameters are set in a **.yaml** file that can then be passed as an argument to the NNS proposal that triggers the SNS launch, both for testing and in production.
+There are some parameters that have to be set and others that are set to a default value but that can also be changed. To make sure that all parameters are set to valid values, that are also consistent with each other, there is a tool to validate a given input file.
```
We don't provide defaults like we did in the last method, all values need to be filled in. We do provide an example one in sns-testing, and we can update it to be like the NNS. Would that be useful to link here?

Also, we do not have the validate command yet, the best means of validating is testing via sns-testing, however we will complete this after we release.

LaraAS replied:
* removed the "defaults" part. I don't think we have to link to it here as we do so a few lines further down.

* just to make sure I get it correctly: when I submit a proposal locally (in testing) then the proposal submission would do some validation, right? but there is just not (yet) an additional validation that I can run just over the yaml file without going through the proposal part? I now refrased vaguely to "test the proposal" and propose that we just add more if we add more validation. 

WDYT? Can we resolve this?

DanielThurau replied:
Confirmed offline. Resolving.

DanielThurau commented on docs/developer-docs/integrations/sns/tokenomics/preparation.md:
```diff
@@ -8,44 +8,30 @@ for examples using the
 ['Creating the configuration file'](https://github.com/dfinity/ic/tree/master/rs/sns/cli#creating-the-configuration-file)
 in the SNS CLI tool.
-The tool is also available through the dfx cache like this: `$(dfx cache show)/sns`**
+The tool is also available through the dfx cache like this: `$(dfx cache show)/sns` **
```
This works too, but we should encourage them to start using dfx extensions instead. 

Let me ask if there is a doc section for the sdk on this tool yet.

LaraAS replied:
That would be great. Could you then maybe add a proposl for what to say here: i.e. how to devs get / create this file?

LaraAS replied:
let's remove this and the bullet point above and just link to the template / example files

LaraAS replied:
linked to the two yaml files and closing this.

DanielThurau commented on docs/developer-docs/integrations/sns/tokenomics/preparation.md:
```diff
@@ -58,18 +44,11 @@ ledger canister such as the token name, token symbol, and the ledger transaction
       swap tokens are reserved for future swaps (separate from the treasury). 
       Note that future swap are not yet designed, but having the tokens reserved
```
I'd cut the part that discusses future swaps.

LaraAS replied:
Good call. Removing

DanielThurau commented on docs/developer-docs/integrations/sns/tokenomics/sns-checklist.md:
```diff
@@ -13,11 +13,12 @@ launch – that is a decision by the NNS DAO. Ultimately, following this guide a
-<!-- NEW: [INFO BOX STYLE] 
+:::info
 The NNS has approved a new flow for how SNSs can be launched that only require one proposal.
 The old flow required two NNS and one SNS proposals.
```
```suggestion
The old flow required two NNS and one SNS proposal.```

LaraAS replied:
I assume this should just about using singular instead of plural. I ignored the ```

DanielThurau commented on docs/developer-docs/integrations/sns/tokenomics/sns-checklist.md:
```diff
@@ -57,46 +58,26 @@ It may not be possible to fully decentralize the dapp at the time of the SNS lau
+The proposal can be created from a YAML configuration file and defines things like the token name, token supply, token distribution, transaction fees, dissolve delays and more. 
 
-Geo-restriction can be used to exclude users with IPs in specific countries with the “restricted_countries” parameter, which takes an array of country codes like this: `restricted_countries: [ "CH" ]` 
+For initializing an SNS for your dapp, the configuration parameters must use the documentation created with the SNS tokenomics tool (section 1.1.5 above) to create the YAML configuration file. The [OpenChat config file](https://github.com/open-ic/open-chat/blob/master/sns/config/sns.yml) is a good example of how to format the file (note: the latest version of SNS may have new/different parameters ...
```
Like in the other comment, lets link the new template file here, or to the `Sns Initial parameters` doc section. The OC is the old config method so I would at the very least not link it, or emphasize that this is the old configuration file method

LaraAS replied:
I added the remark that OC is and old format.

For the other part, would you mind making a suggestion? I don't know if we say enough (yet) in the init parameter part of the docs to clarify all quesitons, but maybe we would afte resolving the conversation above

LaraAS replied:
Added links to yaml template and example and linked to preparation side.

aterga commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -62,74 +56,77 @@ Below, we match the stages of decentralization to the commands that the `sns-tes
         <td>1</td>
         <td>Dapp developers choose the initial parameters of the SNS for a dapp</td>
-        <td rowspan="4"><code>./deploy_sns.sh</code></td>
+        <td><code></code></td>
```
[example_sns_init.yaml](https://github.com/dfinity/sns-testing/blob/main/example_sns_init.yaml)

aterga replied:
```suggestion
        <td><code>[example_sns_init.yaml](https://github.com/dfinity/sns-testing/blob/main/example_sns_init.yaml)</code></td>
```

aterga commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -62,74 +56,77 @@ Below, we match the stages of decentralization to the commands that the `sns-tes
-        <td>Dapp developers submit an SNS proposal to handover control of their dapp to the SNS</td>
-        <td rowspan="2"><code>./register_dapp.sh</code></td>
-        <td rowspan="2"> Potentially many proposals if one's dapp has many canisters</td>
+        <td>(Automated) SNS-W deploys SNS canisters</td>
```
```suggestion
        <td rowspan="5">_No commands required_</td>
```

aterga commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -62,74 +56,77 @@ Below, we match the stages of decentralization to the commands that the `sns-tes
-        <td>0</td>
         <td>Upgrade a yet-to-be-decentralized canister by submitting an SNS proposal that can be voted on using the SNS developer neuron.</td>
-        <td><code>./upgrade_test_canister.sh</code></td>
+        <td><code>TODO</code></td>
```
```suggestion
        <td><code>./upgrade_test_canister.sh</code></td>
```

jessiemongeon1 commented on docs/developer-docs/integrations/sns/tokenomics/preparation.md:
```diff
@@ -8,68 +8,42 @@ for examples using the
+  - **Rules of the swap** such as the minimum number of ICP tokens that the swap must collect to be successful and the maximum number of ICP tokens that it will collect. 
+  - **Conditions of swap participation** such as the definition of a text that must be confirmed by users when participating in the swap and the possibility to define countries from which users cannot participate (by IP addresses).
+  
+- **Fallback controllers of the dapp**: If a decentralization swap fails, for example because the targeted minimum number of ICP tokens could not be collected, the control of the dapp canister(s) is handed back to the principals defined in these parameters. Normally, this will be set to principals owned by the developers that controlled the dapp before the attempt to decentralize it.
```
```suggestion
- **Fallback controllers of the dapp**: if a decentralization swap fails, for example because the targeted minimum number of ICP tokens could not be collected, the control of the dapp canister(s) is handed back to the principals defined in these parameters. Normally, this will be set to principals owned by the developers that controlled the dapp before the attempt to decentralize it.
```

jessiemongeon1 commented on docs/developer-docs/integrations/sns/tokenomics/preparation.md:
```diff
@@ -82,17 +56,10 @@ the dapp's control is handed over to the SNS and during the
-at any time, you must make sure that enough of the initial neurons can vote to reach a decison.**
-See [Stage 1 of the launch](../launching/launch-summary.md) for more details of this. 
+Therefore, **if you want to use SNS proposals at this stage, for example to ensure that the dapp can be upgraded at any time, you must make sure that enough of the initial neurons can vote to reach a decison.**
+See [Stage 1 of the launch](../launching/launch-summary-1proposal.md) for more details of this. 
```
```suggestion
See [stage 1 of the launch](../launching/launch-summary-1proposal.md) for more details of this. 
```

LaraAS commented on docs/developer-docs/integrations/sns/launching/index.md:
```diff
@@ -1,18 +1,13 @@
-<!-- NEW: 
-It starts with a detailed description of all Stages of an SNS launch.
+It starts with a detailed description of all stages of an SNS launch.
 Then, it explains the practical commands & actions to go through SNS launch, in terms of concrete tools and commands that are needed to take a project through all of these stages and launch an SNS for it.
```
I slightly rephrased as I think the "terms" is not something "that is needed". 
Please lt me know if you think more adjustments are needed

LaraAS commented on docs/developer-docs/integrations/sns/launching/index.md:
```diff
@@ -1,18 +1,13 @@
--->
\ No newline at end of file
+Therefore, the developer documentation currently includes pages to describe both the legacy and the new flow.
+The pages [Stages of an SNS launch](./launch-summary-1proposal.md) and [Commands & actions to go through SNS launch](./launch-steps-1proposal.md) define the new, recommended 1-proposal launch flow and how to go through it.
```
I would not rename the second page. Even though the name of the page is still "steps" in the files, I wanted to rename the page where it is visible to the users.
Reason: I think it is unclear what's teh difference between "stages" and "steps". Rather, one page should describe the stages and the other one "how to go through these stages".

LaraAS commented:
Thanks for the review @jessiemongeon1 ! I implemented most of your suggestions in the next commit. 
The only think I don't agree with is the name to refer to the page under launch-steps-1proposal.md. Note that I changed the title of this page to "Commands & actions to go through SNS launch" and we don't refer to this as "steps" anywhere anymore (except in the file name). 
I explained more about the reasoning in one of the remarks above. If you disagree with this, let's discuss!

DanielThurau commented on docs/developer-docs/integrations/sns/testing/testing-locally.md:
```diff
@@ -29,109 +29,112 @@ These scripts have been most tested with a stack that looks like this:
-        <td>Proposal #1 (of 3) is passed or rejected</td>
+        <td>Submit NNS proposal to create SNS</td>
+        <td rowspan="1">[./propose_sns.sh](https://github.com/dfinity/sns-testing/blob/main/propose_sns.sh)</td>
     </tr>
     <tr>
         <td>4</td>
-        <td>Dapp developers trigger the SNS canisters to be created on SNS subnet</td>
+        <td>The NNS proposal is decided</td>
+        <td rowspan="5"><code>(No commands required)</code></td>
+        <td rowspan="5"><code> </code></td>
     </tr>
     <tr>
         <td>5</td>
-        <td>Dapp developers submit an SNS proposal to handover control of their dapp to the SNS</td>
-        <td rowspan="2"><code>./register_dapp.sh</code></td>
-        <td rowspan="2"> Potentially many proposals if one's dapp has many canisters</td>
+        <td>(Automated) SNS-W deploys SNS canisters</td>
     </tr>
         <tr>
         <td>6</td>
-        <td>Proposal #2 (of 3) is passed or rejected</td>
+        <td>(Automated) SNS-W sets SNS root as sole controller of dapp</td>
     </tr>
     <tr>
         <td>7</td>
-        <td>Proposal to start the decentralization swap</td>
-        <td rowspan="2"><code>./open_sns_swap.sh</code></td>
+        <td>(Automated) SNS-W initializes SNS canisters according to settings from Step 1</td>
     </tr>
     <tr>
         <td>8</td>
-        <td>Proposal #3 (of 3) is passed or rejected</td>
+        <td>(Automated) SNS swap starts</td>
+    </tr>
+    <tr>
+        <td> </td>
+        <td>Users participate in the swap</td>
+        <td rowspan="1">[./participate_in_sns_swap.sh](https://github.com/dfinity/sns-testing/blob/main/participate_sns_swap.sh)</td>
+        <td rowspan="1"></td>
     </tr>
```
Unfortunately I don't think linking works in tables :/ 

<img width="1020" alt="Screenshot 2023-08-24 at 2 51 12 PM" src="https://github.com/dfinity/portal/assets/15882011/09d376fd-dbb0-4d89-ac8e-a7afe0013d84">

DanielThurau replied:
Also, it looks like we only have a single entry in the `Notes` section.

LaraAS replied:
Thanks for checking!
* I removed the links
* I think the one note is not pretty, but propose that we leave it as is as we plan to anyways further improve this page after 1proposal release
* I also removed the links from the next table.

DanielThurau commented on docs/developer-docs/integrations/sns/tokenomics/preparation.md:
```diff
@@ -8,68 +8,42 @@ for examples using the
+To make sure that all parameters make sense and are also consistent with each other, it is recommended to test the proposal locally.
 
 **To create the SNS parameter yaml file and validate it, follow the steps
 ['Creating the configuration file'](https://github.com/dfinity/ic/tree/master/rs/sns/cli#creating-the-configuration-file)
 in the SNS CLI tool.
-The tool is also available through the dfx cache like this: `$(dfx cache show)/sns`**
+The tool is also available through the dfx cache like this: `$(dfx cache show)/sns` **
```
**To create the SNS parameter yaml file, copy the template file found [here](https://github.com/dfinity/ic/blob/master/rs/sns/cli/sns_init_template.yaml) and fill in the fields with the desired values. An example configuration file with filled out fields can be found in [sns-testing](https://github.com/dfinity/sns-testing/blob/main/example_sns_init.yaml).**
